### PR TITLE
Replace gfx-descriptor by gpu-descriptor

### PR DESCRIPTION
**Pull request by kvark**
**Connections**
Unblocks #1027
cc @zakarumych 

**Description**
Removes the last link we had between us and gfx-rs.

**Testing**
Tested on wgpu-rs

### CODE REVIEW

File: wgpu-core/src/device/descriptor.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            },
+            count: descriptor_count.storage_buffer_dynamic as _,
+        });
+        ranges.retain(|rd| rd.count != 0);
+
+        match hal::device::Device::create_descriptor_pool(
+            self.0,
+            max_sets as usize,
+            ranges,
+            hal::pso::DescriptorPoolCreateFlags::from_bits_truncate(flags.bits() as u32),
```
Comment (zakarumych) [inline wgpu-core/src/device/descriptor.rs:124]: Even if bits probably will be the same as both gfx and gpu-descriptor tend to use same bits as Vulkan,
it is not guaranteed.

Comment (kvark) [reply wgpu-core/src/device/descriptor.rs:124]: Can we guarantee that? I mean, at least specify in documentation of gpu-alloc and gpu-descriptor that the bits match Vulkan.

Comment (zakarumych) [reply wgpu-core/src/device/descriptor.rs:124]: Done

**Review by cwfitzgerald — APPROVED** [review]
Nothing looks out of place! Very excited to get this going!

**Comment by kvark** [comment]
bors r=cwfitzgerald

**Pull request merged**